src/plugins/intel_gpu/src/graph/space_to_batch.cpp

```diff
@@ -58,6 +60,95 @@ layout space_to_batch_inst::calc_output_layout(space_to_batch_node const& node,
+        auto out_shape = ov::PartialShape::dynamic(input0_layout.get_partial_shape().size());
+        return { layout{out_shape, input0_layout.data_type, format::get_default_format(out_shape.rank().get_length())} };
+    }
+
+
+    ShapeType block_shape = desc->shape_constant == 0 ? impl_param.get_input_layout(1).get<ShapeType>() : ov::Shape{ input0_size };
+    ShapeType begin_shape = desc->shape_constant == 0 ? impl_param.get_input_layout(2).get<ShapeType>() : ov::Shape{ input0_size };
+    ShapeType end_shape = desc->shape_constant == 0 ? impl_param.get_input_layout(3).get<ShapeType>() : ov::Shape{ input0_size };
+
+    ov::op::v1::SpaceToBatch op;
+    std::vector<ShapeType> output_shapes = {ShapeType{}};
+    std::vector<ShapeType> input_shapes = {
+        input0_shape,
+        block_shape,
+        begin_shape,
+        end_shape
+    };
+
+    std::map<size_t, ngraph::HostTensorPtr> const_data;
+    if (desc->shape_constant) {
+        auto block_sizes = tensorToVec(block_data, input0_format);
+        auto begin_sizes = tensorToVec(begin_data, input0_format);
+        auto end_sizes = tensorToVec(end_data, input0_format);
+
+        auto block_values = static_cast<void*>(block_sizes.data());
+        auto begin_values = static_cast<void*>(begin_sizes.data());
+        auto end_values = static_cast<void*>(end_sizes.data());
+
+        auto block_tensor = make_host_tensor({ block_shape, data_types::i32, input0_format }, block_values);
+        auto begin_tensor = make_host_tensor({ begin_shape, data_types::i32, input0_format }, begin_values);
+        auto end_tensor = make_host_tensor({ end_shape, data_types::i32, input0_format }, end_values);
+
+        const_data.emplace(1, block_tensor);
+        const_data.emplace(2, begin_tensor);
+        const_data.emplace(3, end_tensor);
+
+        output_shapes = ov::op::v1::shape_infer(&op, input_shapes, ov::make_tensor_accessor(const_data));
+    } else {
+        auto block_mem = constant_mem.at(1);
+        auto begin_mem = constant_mem.at(2);
+        auto end_mem = constant_mem.at(3);
+
+        cldnn::mem_lock<uint8_t, mem_lock_type::read> lock1(block_mem, impl_param.get_stream());
+        cldnn::mem_lock<uint8_t, mem_lock_type::read> lock2(begin_mem, impl_param.get_stream());
+        cldnn::mem_lock<uint8_t, mem_lock_type::read> lock3(end_mem, impl_param.get_stream());
+
+        auto block_tensor = make_host_tensor(block_mem->get_layout(), lock1.data());
+        auto begin_tensor = make_host_tensor(begin_mem->get_layout(), lock2.data());
+        auto end_tensor = make_host_tensor(end_mem->get_layout(), lock3.data());
+
+        const_data.emplace(1, block_tensor);
+        const_data.emplace(2, begin_tensor);
+        const_data.emplace(3, end_tensor);
+
+        output_shapes = ov::op::v1::shape_infer(&op, input_shapes, ov::make_tensor_accessor(const_data));
+    }
```

Comment:
Is there any possibility that some of block_mem/begin_mem/end_mem are constant and the others are not constant? 
If so and those cases are not handled yet, could you add some exception handling for the unsupported cases?

Reply:
added assert at CreateSpaceToBatchOp and CreateBatchToSpaceOp